model_check.py

```diff
@@ -0,0 +1,751 @@
+        f.write(quant_table)
+        f.close()
+
+        # If run_start is specified this is likely for regression testing
+        # or other debugging.  In this case write out the full predicted and
+        # telemetered dataset as a pickle.
+        if opt.run_start:
+            filename = os.path.join(outdir, 'validation_data.pkl')
+            self.logger.info('Writing validation data %s' % filename)
+            f = open(filename, 'w')
+            pickle.dump({'pred': pred, 'tlm': tlm}, f, protocol=-1)
+            f.close()
+
+        return plots
+
+    def rst_to_html(self, opt, proc):
+        """Run rst2html.py to render index.rst as HTML"""
+
+        # First copy CSS files to outdir
+        import Ska.Shell
+        import docutils.writers.html4css1
+        dirname = os.path.dirname(docutils.writers.html4css1.__file__)
+        shutil.copy2(os.path.join(dirname, 'html4css1.css'), opt.outdir)
+
+        shutil.copy2(os.path.join(TASK_DATA, 'model_check.css'), opt.outdir)
+
+        spawn = Ska.Shell.Spawn(stdout=None)
+        infile = os.path.join(opt.outdir, 'index.rst')
+        outfile = os.path.join(opt.outdir, 'index.html')
+        status = spawn.run(['rst2html.py',
+                            '--stylesheet-path={}'
+                            .format(os.path.join(opt.outdir, 'model_check.css')),
+                            infile, outfile])
+        if status != 0:
+            proc['errors'].append('rst2html.py failed with status {}: see run log'
+                                  .format(status))
+            self.logger.error('rst2html.py failed')
+            self.logger.error(''.join(spawn.outlines) + '\n')
+
+        # Remove the stupid <colgroup> field that docbook inserts.  This
+        # <colgroup> prevents HTML table auto-sizing.
+        del_colgroup = re.compile(r'<colgroup>.*?</colgroup>', re.DOTALL)
+        outtext = del_colgroup.sub('', open(outfile).read())
+        open(outfile, 'w').write(outtext)
+
+    def write_index_rst(self, opt, proc, plots_validation, valid_viols=None,
+                        plots=None, viols=None):
+        """
+        Make output text (in ReST format) in opt.outdir.
+        """
+        # Django setup (used for template rendering)
+        import django.template
+        import django.conf
+        try:
+            django.conf.settings.configure()
+        except RuntimeError, msg:
+            print msg
+
+        outfile = os.path.join(opt.outdir, 'index.rst')
+        self.logger.info('Writing report file %s' % outfile)
+        django_context = django.template.Context(
+            {'opt': opt,
+             'plots': plots,
+             'viols': viols,
+             'valid_viols': valid_viols,
+             'proc': proc,
+             'plots_validation': plots_validation,
+             })
+        index_template_file = ('index_template.rst'
+                               if opt.oflsdir else
+                               'index_template_val_only.rst')
```

Comment:
If the daily stuff is removed, does the "validation only" template and this logic need to persist?  I'm trying to remember the use cases.